Wikipedia/Code/WMFCompassView.m

```diff
@@ -223,4 +223,21 @@ - (void)drawCompassLinesInContext:(CGContextRef)ctx
     }
 }
 
+#pragma mark - Accessibility
+
+- (BOOL)isAccessibilityElement {
+    return YES;
+}
+
+- (NSString *)accessibilityLabel {
+    double positive_angle = self.angleRadians >= 0 ? self.angleRadians : self.angleRadians + 2*3.14;
+    NSInteger clock_direction = lroundf(positive_angle / (3.14 / 6));
+    if (clock_direction == 0) {
+        clock_direction = 12;
+    }
+    NSString* label = MWLocalizedString(@"compass-direction", nil);
+    label = [label stringByReplacingOccurrencesOfString:@"$1" withString:[NSString localizedStringWithFormat:@"%@", @(clock_direction)]];
```

Comment:
Regarding the clock direction system, this is widely used among blind people also outside of English speaking countries. I can definitely confirm that for my country, the Czech Republic. 

It is being used for description of direction in orientation (e.g. when you describe to a blind person where objects are around him), in navigation (e.g. the Blindsquare iOS app) or e.g. to describe the position of items on someones plate while eating.

But Brian is right, we need to check how this is handled in more remote countries like China. I get a feeling that the clockface is very widely known. I will try to check it and I will write another comment here.

I think at least in countries where it is routinely used among blind people, it is much superior to giving degrees in a situation like this. Degrees carry too much detail and are complicated to imagine. I do not know how about you, but I cannot intuitively (without some thinking) realize what a direction like like 210 degrees, 72 degrees or 315 degrees actually mean around me.

The Compass app is not a good example. In that case, the aim is to give an exact direction and that is the principal purpose of the app. In the Wikipedia app, if I understand correctly, the main purpose is giving the users easy access to articles containing information about their surroundings. The information about distance and direction in this later case serve another purpose and should be easily understandable.

So maybe we will have to differentiate based on the current language settings and maybe default to degrees where a better system is not available.

For the same reason, I also think that the cardinal direction is not really necessary, although it could be a nice addition and might also be relevant in some cases.

Regarding the first comment, isn't localizedStringWithFormat necessary for languages which use other characters for numerals, e.g. Japanese?

Thanks for the other comments as well!